src/client/address_book/storage.cc

```diff
@@ -65,18 +67,34 @@ bool AddressBookStorage::GetAddress(
   return true;
 }
 
-void AddressBookStorage::AddAddress(
-    const kovri::core::IdentityEx& address) {
-  auto filename = GetAddressesPath() / (address.GetIdentHash().ToBase32() + ".b32");
-  std::ofstream file(filename.string(), std::ofstream::binary);
-  if (!file)
-    LOG(error) << "AddressBookStorage: can't open file " << filename;
+void AddressBookStorage::AddAddress(const kovri::core::IdentityEx& address)
+{
+  auto filename =
+      GetAddressesPath() / (address.GetIdentHash().ToBase32() + ".b32");
+  kovri::core::OutputFileStream file(filename.string(), std::ofstream::binary);
+  if (!file.Good())
+    {
+      m_Exception.Dispatch(__func__);
+      throw std::runtime_error("failed to open file for writing");
+    }
+  // For sanity, triple-check the validity of the identity length
   const std::size_t len = address.GetFullLen();
-  auto buf = std::make_unique<std::uint8_t[]>(len);
-  // For sanity, the validity of identity length is incumbent upon the parent caller.
-  // TODO(unassigned): triple check that this is the case
-  address.ToBuffer(buf.get(), len);
-  file.write(reinterpret_cast<char *>(buf.get()), len);
+  if (len < kovri::core::DEFAULT_IDENTITY_SIZE
+      || len > kovri::core::MAX_IDENTITY_SIZE)
+    {
+      m_Exception.Dispatch(__func__);
+      throw std::length_error("invalid Identity length");
+    }
+  else
+    {
+      std::vector<std::uint8_t> buf(len);
+      address.ToBuffer(buf.data(), len);
+      if (!file.Write(buf.data(), len))
+        {
+          m_Exception.Dispatch(__func__);
```

Comment:
This is useless because nothing is thrown before it. Exception dispatchers are meant to be used when catching. If you do more than one check, catching would be appropriate. Otherwise, leave it to the caller.